tensorflow/core/util/gpu_solvers.h

```diff
@@ -190,6 +220,43 @@ class CudaSolver {
 
   OpKernelContext* context() { return context_; }
 
+
+#if TENSORFLOW_USE_ROCM
+  // ====================================================================
+  // Wrappers for ROCSolver start here
+  //
+  // The method names below
+  // map to those in ROCSolver, which follow the naming
+  // convention in LAPACK see
+
+  // LU factorization.
+  // Computes LU factorization with partial pivoting P * A = L * U.
+
+  template <typename Scalar>
+  Status Getrf(int m, int n, Scalar* dev_A, int lda, int* dev_pivots, int* info);
+
+  // Uses LU factorization to solve A * X = B.
+  template <typename Scalar>
+  Status Getrs(const rocblas_operation trans, int n, int nrhs, Scalar* A,
+               int lda, const int* dev_pivots, Scalar* B, int ldb);
+
+  template <typename Scalar>
+  Status
+  GetrfBatched(int m, int n, Scalar** dev_A, int lda, int* dev_pivots,
+               rocblas_stride stride, DeviceLapackInfo* info, const int batch_count);
+
+  template <typename Scalar>
+  Status GetrsBatched(const rocblas_operation trans, int n, int nrhs,
+                      Scalar** A, int lda, int* dev_pivots,
+                      rocblas_stride stride, Scalar** B, const int ldb,
+                      const int batch_count);
+
+
+  template <typename Scalar>
+  Status Trsm(rocblas_side side, rocblas_fill uplo, rocblas_operation trans,
+              rocblas_diagonal diag, int m, int n, const Scalar* alpha,
+              const Scalar* A, int lda, Scalar* B, int ldb);
```

Comment:
when you file the upstream PR, see if you can come up with a way to keep the interface/signature for these functions the same for both CUDA and ROCm. My guess is that the upstream reviewers will request for this